fabric-tool-attribute-api-v1/src/main/java/net/fabricmc/fabric/impl/tool/attribute/ToolHandlers.java

```diff
@@ -0,0 +1,343 @@
+package net.fabricmc.fabric.impl.tool.attribute;
+
+import com.google.common.collect.ImmutableList;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.AxeItem;
+import net.minecraft.item.HoeItem;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.item.PickaxeItem;
+import net.minecraft.item.ShearsItem;
+import net.minecraft.item.ShovelItem;
+import net.minecraft.item.SwordItem;
+import net.minecraft.item.ToolItem;
+import net.minecraft.item.ToolMaterial;
+import net.minecraft.item.ToolMaterials;
+import net.minecraft.recipe.Ingredient;
+import net.minecraft.tag.Tag;
+import net.minecraft.util.ActionResult;
+
+import net.fabricmc.api.ModInitializer;
+import net.fabricmc.fabric.api.tool.attribute.v1.DynamicAttributeTool;
+import net.fabricmc.fabric.api.tool.attribute.v1.FabricToolTags;
+import net.fabricmc.fabric.api.tool.attribute.v1.ToolManager;
+import net.fabricmc.fabric.mixin.tool.attribute.MiningToolItemAccessor;
+import net.fabricmc.fabric.mixin.tool.attribute.ToolItemAccessor;
+
+/**
+ * Entrypoint to register the default tool handlers.
+ */
+public class ToolHandlers implements ModInitializer {
+	@Override
+	public void onInitialize() {
+		ToolManager.general().register(new ModdedToolsModdedBlocksToolHandler());
+		ToolManager.general().register(new VanillaToolsModdedBlocksToolHandler());
+		ToolManager.general().register(new VanillaToolsVanillaBlocksToolHandler());
+		ToolManager.tag(FabricToolTags.PICKAXES).register(new ModdedMiningToolsVanillaBlocksToolHandler(
+				ImmutableList.of(
+						createPickaxe(ToolMaterials.WOOD),
```

Comment:
If my two comments below are viable, we should be able to use plain vanilla item instances directly besides removing quite some code.

Reply:
I have changed this to use vanilla tools, only changing the material if needed. Don't know if this is a good approach.

Reply:
This is not feasible since it is changing global state in a multi-threaded environment, causing e.g. integrated server behavior to change while the client thread mines something. Changing material or mining level parameters would first have to be removed.